text/3349-cfg-os-version.md

```diff
@@ -0,0 +1,142 @@
+- Feature Name: `cfg-os-version`
+- Start Date: 2022-10-31
+- RFC PR: TBD
+- Rust Issue: TBD
+
+# Summary
+[summary]: #summary
+
+A new `cfg` key-value option `target_os_version`, and new predicates `os_version_eq`, `os_version_min`, and `os_version_range` that allow users to declare the primary (target-defined) API level required/supported by a block.  A second version of the predicates would take an additional key argument allowing targets to specify the versions of different OS components, e.g. kernel and libc versions.
+
+For instance, the standard library's Windows Mutex implementation could potentially take advantage of this mechanism instead of relying on runtime API detection:
+
+```rust
+pub unsafe fn unlock(&self) {
+    *self.held.get() = false;
+    if cfg!(os_version_min("6.1.7600")) { // `cfg!(os_version_min("Windows7"))` is also possible
```

Comment:
This doesn't match the rest of the RFC where it using the syntax `os_version_min("windows", "6.1.7600")`. Is that a mistake, or is this a different form?

Reply:
Fixed.  Thanks!

Reply:
Just checking, I don't see a new commit pushed. Is that forthcoming?